usr/libexec/greenboot/greenboot

```diff
@@ -7,6 +8,54 @@ SCRIPTS_CHECK_PATHS=("/usr/lib/greenboot/check" "/etc/greenboot/check")
 SCRIPTS_GREEN_PATHS=("/usr/lib/greenboot/green.d" "/etc/greenboot/green.d")
 SCRIPTS_RED_PATHS=("/usr/lib/greenboot/red.d" "/etc/greenboot/red.d")
 
+source_configuration_file() {
+    GREENBOOT_CONFIGURATION_FILE=/etc/greenboot/greenboot.conf
+    if test -f "$GREENBOOT_CONFIGURATION_FILE"; then
+	# shellcheck source=/etc/greenboot/greenboot.conf
+        source $GREENBOOT_CONFIGURATION_FILE
+    fi
+}
+
+source_configuration_file
+function is_disabled {
+    HEALTHCHECK=$1
+    echo "${DISABLED_HEALTHCHECKS,,}" | tr " " '\n' | grep -F -q -x "$HEALTHCHECK"
+}
+
+script_runner () {
+  local scripts_dir=$1; shift
+  local mode=$1; shift
+  local start_msg=$1; shift
+  local required_hc_failed=false
+  echo "$start_msg"
+  for script in $(find "$scripts_dir" -name '*.sh' | sort); do
+    if is_disabled "$(basename "$script")"; then
```

Comment:
I think you need to escape the double quotes surrounding `$script`:
```bash
"$(basename \"$script\")"
```

Reply:
I just tested with this, it prevented the healthcheck from being skipped even if specified in the config. Going to leave as is for now

Reply:
I see a lot of executions with `"$(basename "$script")"` . Why not using a variable to save the value at the begining of the loop and then use that variable?

Even better, why don't print only the script base name in `find` and use the `${script}` variable directly:
```bash
for script in $(find "${scripts_dir}" -name '*.sh' -type f -printf '%f '); do
    if is_disabled "$script"; then
        echo "'${script}' was skipped, as specified in config"
   else
        local rc=0
        systemd-cat -t "${scripts_dir}/${script}" bash "${script}" || rc=$?
...
```

Reply:
I tried to implement this in the most recent updates but it kept failing to boot with these changes. I'm sure it's something I'm doing wrong, I just haven't figured it out yet